presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/ShardCompactionManager.java

```diff
@@ -179,28 +169,38 @@ private void discoverShards()
             if (!metadataDao.isCompactionEnabled(tableId)) {
                 continue;
             }
-
-            Table table = metadataDao.getTableInformation(tableId);
-
-            CompactionSetCreator compactionSetCreator = getCompactionSetCreator(table);
-            Set<ShardMetadata> shards = getFilteredShards(entry.getValue(), table);
-
-            addToCompactionQueue(compactionSetCreator, tableId, shards);
+            Collection<OrganizationSet> compactionSets = createCompactionSets(tableId, entry.getValue());
+            compactionSets.stream()
+                    .forEach(organizer::enqueue);
         }
     }
 
-    private Set<ShardMetadata> getFilteredShards(List<ShardMetadata> shardMetadatas, Table tableInfo)
+    private Collection<OrganizationSet> createCompactionSets(long tableId, Collection<ShardMetadata> tableShards)
     {
-        Set<ShardMetadata> shards = shardMetadatas.stream()
+        Table tableInfo = metadataDao.getTableInformation(tableId);
+        OptionalLong temporalColumnId = tableInfo.getTemporalColumnId();
+        if (temporalColumnId.isPresent()) {
+            TableColumn tableColumn = metadataDao.getTableColumn(tableId, temporalColumnId.getAsLong());
+            if (!isValidTemporalColumn(tableId, tableColumn.getDataType())) {
+                return EMPTY_SET;
```

Comment:
Also, the preferred way to use the JDK versions is with the functions like `emptySet()` (the constant is a raw generic type)